src/entry.rs

```diff
@@ -0,0 +1,62 @@
+use std::{
```

Comment:
Great that you moved this to a new module. Maybe it should be called `dir_entry`?